ethcore/src/client/client.rs

```diff
@@ -652,10 +651,7 @@ impl Client {
 
 		// Commit results
 		let receipts = block.receipts().to_owned();
-		let traces = block.traces().clone().unwrap_or_else(Vec::new);
-		let traces: Vec<FlatTransactionTraces> = traces.into_iter()
-			.map(Into::into)
-			.collect();
+		let traces = block.traces().clone().drain();
```

Comment:
Seems like unnecessary clone, since `block` is discarded anyway.

Reply:
totally agree!

But for now, we can't omit `clone()`, cause `block` is generic over `IsBlock` trait, and `IsBlock` has only `fn traces(&self)` method.

Reply:
that's why I wrote, that in next pr 'I'll try to reduce number of clones we make during block verification' ;)